### ISSUE TITLE
Question about multiple renders

### ISSUE THREAD

**Issue by Spope** [ficusjs/ficusjs#5]
Hi,

TLDR; I have multiple renders and deleted element re-renders.

I have an application with an app state, containing an array of ids, and a lookup object containing data objects by the same ids. I created 2 customElement, the first iterate on the array of ids, passing it as a prop to a second customElement. This second retrieve the data object from the state to display it. 
The display is correct, everything looks good, but I saw that each element is rendered multiple times and I don't understand why.
A problem appear if I remove some items. To keep my state clean I remove the id from the array and I remove the object by its ID. This is working except for the last item of the list. After being removed from the state, the last item is renderd once more, and the object is not found into the state anymore, throwing an error.

Sorry for the length of this, here is a [codepen](https://codepen.io/Spope/pen/yLMGjVy?editors=1011) reproducing this. If you delete the first or the second item it works (except for the multiple renders) but if you remove the last item, an error is thrown.

Maybe I haven't done things the right way but could not find what.

**Comment by ducksoupdev** [comment]
Thanks for the codepen @Spope , I'll investigate the issue and get back to you.

**Comment by ducksoupdev** [comment]
@Spope apologies for the delay in responding. I have investigated the issue and the elements are only rendering once - I have debugged it through the source code to verify. The renderer does need to re-evaluate the new HTML against the existing rendered DOM when you remove an element so I added a check for the element name prop against the state which removed the error you encountered.

In `element-component`:

```js
createCustomElement(
  'element-component',
  withStore(store, {
    renderer,
    props: {
      elementName: { type: String }
    },
    deleteElement () {
      this.store.deleteElement(this.props.elementName)
    },
    render () {
      console.log(`Render element ${this.props.elementName}`)
      if (!this.store.state.elementsByName[this.props.elementName]) {
        return
      }
      return html`
        <p>${this.store.state.elementsByName[this.props.elementName].name} <button onclick=${this.deleteElement}>X</button></p>
      `
    }
  })
)
```

**Comment by Spope** [comment]
Hey, no need to apologize, thanks for answering ! That was the solution I used Indeed. I wondered why the renderer needed to reevaluate the template. I will try to understand this from the source. Thanks again for your answer and this lib, it's really great to use.
